src/components/SignInForm/SignInForm.jsx

```diff
@@ -0,0 +1,168 @@
+import { useContext, useState } from "react";
+import styles from "./SignInForm.module.scss";
+import { AuthContext } from "../../containers/AuthContext";
+
+const SignInForm = () => {
+  const authContext = useContext(AuthContext);
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const [errors, setErrors] = useState({});
+
+  const handleEyeButtonClick = (e) => {
+    e.preventDefault();
+    setIsPasswordVisible(!isPasswordVisible);
+  };
+
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+    setErrors((prevErrors) => ({ ...prevErrors, email: null }));
+  };
+
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value);
+    setErrors((prevErrors) => ({ ...prevErrors, password: null }));
+  };
+
+  const validateEmail = (email) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!emailRegex.test(email)) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        email: { valid: false, message: "Не правильна адреса" },
+      }));
+      return false;
+    }
+    setErrors((prevErrors) => ({
+      ...prevErrors,
+      email: { valid: true },
+    }));
+    return true;
+  };
+
+  const validatePassword = (password) => {
+    if (password.length < 8) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        password: {
+          valid: false,
+          message:
+            "Не меньше 8 символів",
+        },
+      }));
+      return false;
+    }
+    setErrors((prevErrors) => ({
+      ...prevErrors,
+      password: { valid: true },
+    }));
+    return true;
+  };
+
+  const handleSignIn = async (event) => {
+    event.preventDefault();
+    setIsSubmitting(true);
+
+    if (validateEmail(email) && validatePassword(password)) {
```

Comment:
i think validation should be already done by this time.
coz we validate each input on its 'blur' event, right?
so, here we should rather check for any errors ;)

Reply:
fixed